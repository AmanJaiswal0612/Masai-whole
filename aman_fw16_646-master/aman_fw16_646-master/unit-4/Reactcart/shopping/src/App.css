
.sidebar{
    background-color: gainsboro;
    position:-webkit-sticky;
    position:sticky;
    top:0;
    height: 700px;
    padding: 30px;
}
.sidebar>img{
    width:120px;
    height: 50px;
    margin-bottom: 70px;
}
.sidebar>h2{
    font-weight:400;
}
.sidebar>h2:nth-child(7){
    margin-bottom: 70px;
}
.sidebar>h3{
    color: gray;
    font-weight: lighter;
}
.main{
    display: grid;
    grid-template-columns: 17% 83%;
    
}
.link{
    text-decoration: none;
}

.cart{
    display:flex;
    
}