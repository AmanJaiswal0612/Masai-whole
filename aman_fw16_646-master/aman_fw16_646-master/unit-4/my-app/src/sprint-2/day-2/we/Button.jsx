import React, { useState } from 'react'

function Button() {
const [count,setCount]= useState(0)
 
  return (
    <>
    <button onClick={()=>setCount(count+1)}>Add</button>
    <button onClick={()=>setCount(count-1)}>Minus</button>
    <h1>Count is:{count}</h1>
    </>
  )
}

export default Button