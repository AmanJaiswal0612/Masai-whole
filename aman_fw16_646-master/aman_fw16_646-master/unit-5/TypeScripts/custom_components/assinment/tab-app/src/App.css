.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

button {
  font-size: calc(10px + 2vmin);
}


.tab{
  display: flex;
  width: 500px;
  justify-content: space-around;
  margin: auto;
  border: 1px solid black;
  font-weight: 700;
  font-size: 20px;
}

.pannel{
  width: 500px;
  margin: auto;
  text-align: center;
}
.pannel>div{
  display: none;
}
.pannel>div:first-child{
  display: block;
}