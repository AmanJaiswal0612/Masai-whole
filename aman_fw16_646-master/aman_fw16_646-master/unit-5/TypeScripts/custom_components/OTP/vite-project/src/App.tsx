
import './App.css'
import {OtpInput} from "./components/OtpInput"
import { accdata } from './accdata'
import Accodian from './components/Accodian'

export default function App() {
  const handleChange= (otp:string)=>{
    console.log(otp)
  }


   
  return (
    <div className="App">
       <OtpInput onChange={handleChange} length={4} />
       <Accodian data={accdata}/>

    </div>
  );
}
