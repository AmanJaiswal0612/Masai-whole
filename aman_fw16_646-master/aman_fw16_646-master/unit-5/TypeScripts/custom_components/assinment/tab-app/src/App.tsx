import { useState } from 'react'
import './App.css'
import Tab from './components/Tab'
import TabBox from './components/TabBox';

type TheadProps={
  val:string[];
} 

function App() {
  
  
  return (
    <div className="App">
        <TabBox/>
    </div>
  )
}

export default App
