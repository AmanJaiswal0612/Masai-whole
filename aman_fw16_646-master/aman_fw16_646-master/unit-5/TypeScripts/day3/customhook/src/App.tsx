import React from 'react';
import logo from './logo.svg';
import './App.css';
import GithubSearch from './components/GithubSearch';
import GTimer from './components/GTimer';
import Form from './components/Form';
import Demo from './components/Demo';
import CountrySearch from './components/CountrySearch';

function App() {
  return (
    <div className="App">
      {/* <GithubSearch/> */}
      {/* <GTimer/> */}
      {/* <Form/> */}
      {/* <Demo/> */}
      <CountrySearch/>
    </div>
  );
}

export default App;
