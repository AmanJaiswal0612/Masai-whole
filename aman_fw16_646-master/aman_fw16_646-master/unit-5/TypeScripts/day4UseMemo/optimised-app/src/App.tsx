import { nanoid } from 'nanoid';
import React, { useCallback, useEffect, useRef, useState } from 'react';
import './App.css';
import {MemoForm} from './components/Form';
import { TaskObj } from './components/Form';
import {MemoShow} from './components/Show';

function App() {
 const[count,setCount]=useState<number>(0);
 const[data,setData]= useState<TaskObj[]>([])
 let timerId:any= useRef();
useEffect(()=>{
 console.log("inside use effect")
 timerId.current=setInterval(()=>{
       setCount((prev)=>{
         return prev+1;
       })
 },2000)
},[])



const addTask=(payload:TaskObj)=>{
  console.log(payload)
 setData([...data,payload])
}

const cb = useCallback(addTask,[data])

const toogleVerify= (id?:string)=>{
   let ndata=data.map((el)=>{
     return (el.id===id?{...el,verify:!el.verify}:el)
   })
   setData(ndata)
}

const tooglecb = useCallback(toogleVerify,[data])

  return (
    <div className="App">
      <h1>Count:{count}</h1>
      <div>
        <MemoForm addTask={cb} />
        <div id="box">
        {data.map((el)=>{
          return <MemoShow key={el.id} toggleVerify={tooglecb} {...el} />
        })}
        </div>
      </div>
    </div>
  );
}

export default App;
