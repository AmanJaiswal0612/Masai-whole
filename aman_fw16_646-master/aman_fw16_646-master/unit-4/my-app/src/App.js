
import './App.css';
import Assin1 from './sprint-1/day-1/assignments/Assin1';
import { nanoid } from 'nanoid'
import Commonbtn from './sprint-1/day-1/assignments/Commonbtn';
import styles from "./sprint-1/day-1/assignments/Commonbtn.module.css"
import { Counter } from './sprint-1/day-2/assignments/Counter';
import { Todo } from './sprint-1/day-2/assignments/Todo';
import { PaymentCard } from './sprint-1/day-3/assignments/PaymentCard';
import TodoList from './sprint-1/day-3/we/TodoList';
import { ParentToChild } from './sprint-2/day-1/we/ParentToChild';
import { ChildToParent } from './sprint-2/day-1/we/ChildToParent';
import { Sibling } from './sprint-2/day-1/we/Sibling';
import { NewTodo } from './sprint-2/day-1/we/NewTodo';

import { Tictac } from './sprint-2/day-1/assignments/Tictac';
import NewCount from './sprint-2/day-2/we/NewCount';
import TodoA from './sprint-2/day-2/we/TodoA';
import WTodo from './sprint-2/day-3/we/WTodo';
import Timer from './sprint-2/day-3/assignments/Timer';
import DForm from './sprint-2/day-4/assignments/DForm';
import EmployDetails from './sprint-2/day-4/assignments/EmployDetails';



const btnData= ["Join US","Setting","Login","Contact Us","Search","Help","Home","Download"]


function App() {
  return (
    <>
      {/* <Assin1/>
      <div className={styles.commonbtn}>
          {btnData.map((el)=>{
           return <Commonbtn  key={nanoid()} name={el}/>
          })}
      </div> */}
      {/* <Counter no={0}/>
      <Todo/>
      <PaymentCard product={`Desktop  -   Mobile`} pay={`Pay`} gift={`Amazon Gift`} color={"orange"} date={"28/10/2021"} logo={"https://th.bing.com/th/id/OIP.GOGAIYvawWhbeRl8kCMlmwHaHa?pid=ImgDet&rs=1"}/>
     <PaymentCard product={`Mac Os  -   Mobile`} pay={`Payment`} gift={`Apple Gift`} color={`lightgrey`} date={"17 sep 2021"} logo={`https://th.bing.com/th/id/R.148b09308201fee731f0acf73b50b902?rik=DVbEhjltDtg5wA&riu=http%3a%2f%2f1000logos.net%2fwp-content%2fuploads%2f2016%2f10%2fApple-Logo.png&ehk=AhTd7sHYgkILUL4whTrdx5yMkoOAP1lBEEnS0YlzSr4%3d&risl=&pid=ImgRaw&r=0`}/> 
     <TodoList/>
     <ParentToChild/> 
      <ChildToParent/> 
     <Sibling/> 
      <NewTodo/>
     <Tictac/>
     <NewCount/>
     <TodoA/> */}
     {/* <Res/> */}
     {/* <WTodo/> */}
     {/* <Timer start={15} end={5} time={1000}/> */}
     {/* <DForm></DForm> */}
     {/* <EmployDetails/> */}
     <WTodo/>
    </>
  );
}

export default App;
