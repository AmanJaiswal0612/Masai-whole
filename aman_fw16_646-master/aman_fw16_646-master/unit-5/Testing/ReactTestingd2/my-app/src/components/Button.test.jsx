
import {fireEvent, getByTestId, render,screen} from "@testing-library/react"
import App from "../App"
import Button from "./Button"
import renderer from "react-test-renderer"


describe("Button test", ()=>{

    it("should have button in dom", ()=>{
      render(<Button>Click Me</Button>)


      let button = screen.getByText("Click Me");
      expect(button).toBeInTheDocument();
    })

    it("should have button id",()=>{
        render(<Button>Click Me</Button>)

        let button = screen.getByTestId("cButton");
        expect(button).toBeInTheDocument();
    })

    it("check empty button",()=>{
        render(<Button></Button>);

        let button = screen.getByTestId("cButton");
        expect(button).toBeEmptyDOMElement();
    })


    it("should change theme",()=>{
        render(<App></App>)

        let h3= screen.getByText("Theme is light");
        expect(h3).toHaveTextContent("light");

        let themeButton=screen.getByText("change theme");
        fireEvent.click(themeButton);
        expect(h3).toHaveTextContent("dark")
    })

    it("should call given function", ()=>{
        const mockfn= jest.fn();
        render(<Button onClick={mockfn} >Click Me</Button>);

        const btn= screen.getByText("Click Me");
        fireEvent.click(btn);
        fireEvent.click(btn);

        expect(mockfn).toBeCalledTimes(2)
    })

    it("Should match snapshot",()=>{
        const tree= renderer.create(
            <Button colorScheme={"red"} variant="ghost">Add Counter</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot()
    })

   

})