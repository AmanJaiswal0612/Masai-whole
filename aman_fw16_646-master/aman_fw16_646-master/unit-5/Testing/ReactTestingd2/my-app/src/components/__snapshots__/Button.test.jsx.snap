// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Should match snapshot 1`] = `
<button
  className="btn red ghost"
  data-testid="cButton"
>
  Add Counter
</button>
`;
